landingai/pipeline/frameset.py

```diff
@@ -151,6 +152,41 @@ def to_numpy_array(self, image_src: str = "") -> np.ndarray:
         img = self.image if image_src == "" else self.other_images[image_src]
         return np.asarray(img)
 
+    def show_image(
+        self, image_src: str = "", clear_nb_cell: bool = False, *, include_predictions: bool = False
+    ) -> "FrameSet":
+        """Open an a window and display all the images.
+        Parameters
+        ----------
+        image_src (deprecated): if empty the source image will be displayed. Otherwise the image will be selected from `other_images`
+        include_predictions: If the image has predictions, should it be overlayed on top of the image?
+        """
+        if image_src:
+            warnings.warn(
+                "image_src keyword on Frame.show_image is deprecated. Use include_predictions instead."
+            )
```

Comment:
Wow... this is a more severe change..... there many decisions that are based on assumptions about how the SDK would evolve... supporting "versions of an image" is one..... in past project like zoomlion for debugging purposes each step on the pipeline would save a copy of the image.... anyway I don't mind the `include_predictions` shortcut, I'm more doubt full about the warning... 

BTW: Many of the design decisions and background is documented here.... we can go over it if you want..
https://docs.google.com/document/d/12fAtcD9YjcGX4NHANNOvi7d0pLpublC216d8i1wCZaY/edit

Reply:
There other places where you would need `include_predictions` like `to_numpy_array` ..... can you think of a less restrictive way to expose this idea of image variants?? I like the idea of having a construct like `to_numpy_array` that the compiler can type check, but it is too restrictive... :(

Reply:
Which other variants would we include for now, apart from overlay predictions? I don't understand what you mean by versioning a single Frame.

It is not clear what an `image_src` is for the end user of the SDK, and it shows when writing the documentation (I would need one more paragraph about what an `image_src` is, instead of just presenting a new boolean). If we only have one concrete use case for `image_src` (that is: prediction overlay), we should prefer the simple boolean instead of a more complex interface to decide to include or not that prediction overlay.

TBH, I even thought about always saving the image with overlay if the prediction is present, and just providing a `frame.copy()` method to allow the user to have the original `Frame` before running inference. It is intuitive, and supports the use cases you're mentioning.

About the other methods, we should add the warning before bumping version (maybe in a new PR?). But I didn't add it now because I didn't even pass by them while writing the documentation. As a user integrating common production use cases with the SDK, I didn't have a single concrete use case to convert a frame to numpy (although I understand it might be needed for more advanced usage). I have a strong feeling that if I have this need, it will be mostly because we are lacking something that we should improve in the SDK itself.